source/loader/ze_loader.cpp

```diff
@@ -135,6 +135,47 @@ namespace loader
         return flags_value;
     }
 
+    bool driverSortComparitor(const driver_t &a, const driver_t &b) {
+        bool pciOrderingRequested = getenv_tobool( "ZE_ENABLE_PCI_ID_DEVICE_ORDER" );
```

Comment:
in case of ZE_ENABLE_PCI_ID_DEVICE_ORDER set I believe that all runtimes should also respect the flag and expose lowest pci id first, so we can store pci_id of first device of each runtime and the sort by that

Reply:
Fixed in latest, the comparator function now reads the driver's pci device order setting such that the env is only read one time and then the value is set for all drivers.